go/vt/vtadmin/errors/typed_error.go

```diff
@@ -77,3 +78,29 @@ func (e *MissingParams) Error() string {
 func (e *MissingParams) Code() string         { return "missing params" }
 func (e *MissingParams) Details() interface{} { return nil }
 func (e *MissingParams) HTTPStatus() int      { return 400 }
+
+// NoSuchSchema is returned when a schema definition cannot be found for a given
+// set of filter criteria. Both GetSchema and FindSchema can return this error.
+type NoSuchSchema struct {
+	Clusters []string
+	Table    string
+}
+
+func (e *NoSuchSchema) Error() string {
+	return fmt.Sprintf("%s: no schemas found with table named %s", e.Code(), e.Table)
+}
+
+func (e *NoSuchSchema) Details() interface{} {
+	details := map[string]interface{}{
+		"table": e.Table,
+	}
+
+	if e.Clusters != nil {
+		details["clusters"] = e.Clusters
+	}
+
+	return details
+}
+
+func (e *NoSuchSchema) Code() string    { return "no such schema" }
+func (e *NoSuchSchema) HTTPStatus() int { return http.StatusNotFound }
```

Comment:
Open TODO for anyone to swap the rest of our magic numbers in this file to reference the constants in `net/http`

Reply:
Do you mean in this PR, or a separate one?

Reply:
Separate one! It's not really related to this change, but something we should do. https://github.com/vitessio/vitess/projects/12#card-59345039